packages/nodes-base/nodes/NocoDB/GenericFunctions.ts

```diff
@@ -35,14 +35,28 @@ interface IAttachment {
  * @returns {Promise<any>}
  */
 export async function apiRequest(this: IHookFunctions | IExecuteFunctions | ILoadOptionsFunctions | IPollFunctions, method: string, endpoint: string, body: object, query?: IDataObject, uri?: string, option: IDataObject = {}): Promise<any> { // tslint:disable-line:no-any
-	const credentials = await this.getCredentials('nocoDb');
+	const authenticationMethod = this.getNodeParameter('authentication', 0);
+	let credentials;
+
+	if (authenticationMethod === 'nocoDbApiToken') {
+		credentials = await this.getCredentials('nocoDbApiToken');
+	} else if (authenticationMethod === 'nocoDb') {
+		credentials = await this.getCredentials('nocoDb');
+	}
```

Comment:
Can we not simply do the following here?
`credentials = await this.getCredentials(authenticationMethod);`

Reply:
Changed 👍🏻